### ISSUE TITLE
Using a custom OAuth provider

### ISSUE THREAD

**Issue by domurtag** [alvarosanchez/grails-spring-security-rest#234]
Currently I use the OAuth delegation feature provided by the plugin with the Facebook and Twitter providers. I would like to add another OAuth provider, which is not supported by Pac4J, but I'm not sure if this is possible?

Would you expect the following to work?

- Add provider configuration to `Config.groovy`

````groovy
grails.plugin.springsecurity.rest.oauth {

  // facebook and google providers configured here

  custom {
    client = org.example.CustomClient
    scope = 'scope_name'
    defaultRoles = ['ROLE_USER']
  }
}
````

- Implement `org.example.CustomClient` (probably by subclassing `BaseOAuth20Client`)

Authentication with this provider would be triggered by a request `GET /oauth/authenticate/custom`

**Comment by domurtag** [comment]
I tried this approach and it worked like a charm. You have to implement a few other classes besides `CusomClient`, e.g. `CustomApi`, `CustomOAuth20ServiceImpl`, `CustomProfile`, but there are plenty of examples in Scribe that you can lean on.

Is it worth adding a note to the docs explaining how to add a custom OAuth provider or at least mentioning that it's possible?

**Comment by alvarosanchez** [comment]
Fixed by e3aca8b04ceade35334b2688a444844f8eb10234
